lib/_stream_wrap.js

```diff
@@ -47,12 +44,12 @@ function StreamWrap(stream) {
     self.emit('error', err);
   });
   this.stream.on('data', function ondata(chunk) {
-    if (!(chunk instanceof Buffer)) {
+    if (typeof chunk === 'string' || this._readableState.objectMode === true) {
       // Make sure that no further `data` events will happen
       this.pause();
       this.removeListener('data', ondata);
 
-      self.emit('error', new errors.Error('ERR_STREAM_HAS_STRINGDECODER'));
+      self.emit('error', new errors.Error('ERR_STREAM_WRAP'));
```

Comment:
Well, the error code was very specific and somewhat misleading that `objectMode` would or should not trigger this even though `objectMode` always would have resulted in a misleading error since it's first existence.

I think now it better represents both cases. I do see that it's not ideal to rename the code though.